### ISSUE TITLE
.removeItem() interferes with auto-range

### ISSUE THREAD

**Issue by psyser** [domarm-comat/pglive#17]
I can't tell if the problem is caused by pyqtgraph or pglive.

When I remove a plot with foo.removeItem(bar), it doesn't seem to be completely removed, so the auto-range display no longer works.

My condition is that the second plot is included in the auto-range as soon as it appears and excluded as soon as it is removed. This doesn't seem to work and I can't find a way to solve this problem.

I took the examples code and modified it slightly so that the problem is reproducible:

```
import sys
from math import sin
from threading import Thread
from time import sleep

from PySide6.QtWidgets import QApplication

from pglive.sources.data_connector import DataConnector
from pglive.sources.live_plot import LiveLinePlot
from pglive.sources.live_plot_widget import LivePlotWidget

"""
In this example Line plot is displayed.
"""
app = QApplication(sys.argv)
running = True

plot_widget = LivePlotWidget(title="Line Plot @ 100Hz")
plot_curve = LiveLinePlot()
plot_widget.addItem(plot_curve)
# DataConnector holding 600 points and plots @ 100Hz
data_connector = DataConnector(plot_curve, max_points=600, update_rate=100)

plot_curve2 = LiveLinePlot()
data_connector2 = DataConnector(plot_curve2, max_points=600, update_rate=100)
plot_widget.addItem(plot_curve2)

def sin_wave_generator(connector):
    """Sine wave generator"""
    x = 0
    while running:
        x += 1
        data_point = sin(x * 0.01)
        # Callback to plot new data point
        connector.cb_append_data_point(data_point, x)

        sleep(0.01)

def add_and_remove():
  # Wait two seconds before starting
  sleep(2)
  # Generate the second sine wave for three seconds
  x2 = 0
  while x2 <= 300:
    x2 += 1
    data_point = 2+sin(x2 * 0.01)
    # Callback to plot new data point
    data_connector2.cb_append_data_point(data_point, x2)
    sleep(0.01)
  # Remove the seconds sine wave
  plot_widget.removeItem(plot_curve2)
    
plot_widget.show()
Thread(target=sin_wave_generator, args=(data_connector,)).start()
Thread(target=add_and_remove, args=()).start()
app.exec()
running = False
```

**Comment by domarm-comat** [comment]
Thanks for trying pglive and reporting the bug.
This bug is related to pglive and how `live_axis_range` is producing final x and y range for given datapoints.
When plot was removed, data_connector was still present in `live_axis_range` and therefore was interfering with correct range calculation.

I've released new version - 0.6.2 which should fix this problem.

One more thing. I see, that in your example you are adding and removing plot from another Thread. You should not do this as it might cause unexpected crash. You should always use signals for that.

I modified your example to do what you want using slots and overriding PlotLiveWidget.

```
import sys
from math import sin
from threading import Thread
from time import sleep

from PySide6.QtWidgets import QApplication

from pglive.sources.data_connector import DataConnector
from pglive.sources.live_plot import LiveLinePlot
from pglive.sources.live_plot_widget import LivePlotWidget
from pyqtgraph.Qt import QtCore  # type: ignore

"""
In this example Line plot is displayed.
"""
app = QApplication(sys.argv)
running = True

class LivePlotWidgetMod(LivePlotWidget):

    sig_add_plot = QtCore.Signal(object)
    sig_del_plot = QtCore.Signal(object)
    
    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)
        self.sig_add_plot.connect(self.addItem)
        self.sig_del_plot.connect(self.removeItem)

plot_widget = LivePlotWidgetMod(title="Line Plot @ 100Hz")
plot_curve = LiveLinePlot()
plot_widget.addItem(plot_curve)
# DataConnector holding 600 points and plots @ 100Hz
data_connector = DataConnector(plot_curve, max_points=600, update_rate=100)

plot_curve2 = LiveLinePlot()
data_connector2 = DataConnector(plot_curve2, max_points=600, update_rate=100)
plot_widget.addItem(plot_curve2)


def sin_wave_generator(connector):
    """Sine wave generator"""
    x = 0
    while running:
        x += 1
        data_point = sin(x * 0.01)
        # Callback to plot new data point
        connector.cb_append_data_point(data_point, x)

        sleep(0.01)


def add_and_remove():
    # Wait two seconds before starting
    sleep(2)
    # Generate the second sine wave for three seconds
    x2 = 0
    while x2 <= 300:
        x2 += 1
        data_point = 2 + sin(x2 * 0.01)
        # Callback to plot new data point
        data_connector2.cb_append_data_point(data_point, x2)
        sleep(0.01)
    # Remove the seconds sine wave
    plot_widget.sig_del_plot.emit(plot_curve2)


plot_widget.show()
Thread(target=sin_wave_generator, args=(data_connector,)).start()
Thread(target=add_and_remove, args=()).start()
app.exec()
running = False
```

**Comment by psyser** [comment]
Thanks for the immediate update and for improving my code example. I installed the new version and now it works exactly as it should. :-)
